src/index/index.c

```diff
@@ -0,0 +1,141 @@
+#include "index.h"
+
+int compareNodes(const void *a, const void *b) {
+  return ((Node*)a)->id - ((Node*)b)->id;
+}
+
+int compareStrings(void *a, void *b, void *ctx) {
+  return strcmp(((SIValue*)a)->stringval, ((SIValue*)b)->stringval);
+}
+
+// TODO this seems inefficient
```

Comment:
I don't think this is even a safe comparison method, actually, so I will probably replace it with something that is more aware of the numeric types it is comparing soon.